app/src/main/java/com/group80/uoftinder/CreateAccountView.java

```diff
@@ -0,0 +1,264 @@
+
+                        createBasicInfoView(currentUser);
+                    }
+                    else {
+                        Toast.makeText(CreateAccountView.this, "Registration failed :(",
+                                Toast.LENGTH_SHORT).show();
+                    }
+                }
+            });
+        }
+    }
+    /*
+     * Creates the basicinfoview.xml and gets inputs, proceeds to next page if information is
+     * entered correctly
+     */
+    private void createBasicInfoView(User currentUser) {
+
+        setContentView(R.layout.basicinfoview);
+
+        Button enter = findViewById(R.id.cont);
+        enter.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                String userName = ((EditText) findViewById(R.id.name)).getText().toString().trim();
+                String userAge = ((EditText) findViewById(R.id.age)).getText().toString();
+
+                ChipGroup identity_group = findViewById(R.id.identityGroup);
+                int identity_count = identity_group.getChildCount();
+                String identity = "";
+
+                //loops through all chips(answers) of identity answers and finds which chip was
+                //selected
+                for(int i = 0; i<identity_count; i++) {
+                    Chip chip  = (Chip) identity_group.getChildAt(i);
+                    if(chip.isChecked()) {
+                        identity = chip.getText().toString();
+                    }
+                }
+                //loops through all chips(answers) of type answers and finds which chip was
+                //selected
+                ChipGroup type_group = findViewById(R.id.typeGroup);
+                int type_count = type_group.getChildCount();
+                String type = "";
+                for(int i = 0; i<type_count; i++) {
+                    Chip chip  = (Chip) identity_group.getChildAt(i);
+                    if(chip.isChecked()) {
+                        type = chip.getText().toString();
+                    }
+                }
+
+                //checks if all information was entered correctly
+                boolean move_on = control.newAccount(userName, userAge, identity, type);
+                if(move_on) {
+                    //sets information for the currentUser
+                    currentUser.setName(userName);
+                    currentUser.setAge(Integer.parseInt(userAge));
+                    currentUser.setGender(identity);
+                    currentUser.setUserType(type);
+
+                    createQuestionnaireView(currentUser, "Academic");
+                }
+                else {
+                    String text = "Please enter your information correctly";
+                    TextView error = findViewById(R.id.error_basicinfo);
+                    error.setText(text);
+                }
+
+            }
+        });
+
+    }
```

Comment:
Since these lines define a new view with its onw logic, would it be better to create another file `BasicInfoView.java` and move them there? You can then use [`Intent`](https://developer.android.com/reference/android/content/Intent) to switch to the other view.

Reply:
Yeah for the sake of getting all the code working in time for Milestone 4, I shoved these all into one class. However, I will definitely be creating more classes for each view and switching between them using Intent.